### PR TITLE
[zh] Add PodAffinity to API reference for Pod

### PR DESCRIPTION
**Pull request by weiy730**
Signed-off-by: weiy730 <weiy730@163.com>

(#37226 ) Pod article missing PodAffinity

### CODE REVIEW

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 1]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
 
       按节点字段列出的节点选择器要求列表。
 
+## PodAffinity {#PodAffinity}
+
+<!--
+Pod affinity is a group of inter pod affinity scheduling rules.
+-->
+Pod 亲和性是一组 Pod 亲和性调度规则。
```
Comment (windsonsea) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3656]:
```suggestion
Pod 亲和性是一组 Pod 间亲和性调度规则。
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 2]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+
+  The scheduler will prefer to schedule pods to nodes that satisfy the affinity expressions specified by this field, but it may choose a node that violates one or more of the expressions. The node that is most preferred is the one with the greatest sum of weights, i.e. for each node that meets all of the scheduling requirements (resource request, requiredDuringScheduling affinity expressions, etc.), compute a sum by iterating through the elements of this field and adding "weight" to the sum if the node has pods which matches the corresponding podAffinityTerm; the node(s) with the highest sum are the most preferred.
+
+  <a name="WeightedPodAffinityTerm"></a>
+  *The weights of all of the matched WeightedPodAffinityTerm fields are added per-node to find the most preferred node(s)*
+-->
+
+- **preferredDuringSchedulingIgnoredDuringExecution** ([]WeightedPodAffinityTerm)
+
+  调度程序会更倾向于将 Pod 调度到满足该字段指定的亲和性表达式的节点，
```
Comment (windsonsea) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3671]:
```suggestion
  调度器更倾向于将 Pod 调度到满足该字段指定的亲和性表达式的节点，
```
This page has 10 调度器 and 2 调度程序. It is recommended to use the higher hits: 调度器.

Comment (weiy730) [reply content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3671]: OK.

**Comment by sftim** [comment]
/retitle [zh] Add PodAffinity to API reference for Pod

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 3]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+
+  调度器会更倾向于将 Pod 调度到满足该字段指定的亲和性表达式的节点，
+  但它可能会选择违反一个或多个表达式的节点。最优选的节点是权重总和最大的节点，
+  即对于满足所有调度要求（资源请求、`requiredDuringScheduling` 亲和表达式等）的每个节点，
+  通过迭代该字段的元素来计算总和，如果节点具有与相应 `podAffinityTerm`
+  匹配的 Pod，则将 "权重" 添加到总和中； 
+  具有最高总和的节点是最优选的。
+
+  <a name="WeightedPodAffinityTerm"></a>
+  **所有匹配的 WeightedPodAffinityTerm 字段的权重都是按节点添加的，以找到最优选的节点。**
```
Comment (tengqm) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3679]:
```suggestion
  **所有匹配的 WeightedPodAffinityTerm 字段的权重都是按节点累计的，以找到最优选的节点。**
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 4]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+
+    Required. A pod affinity term, associated with the corresponding weight.
+
+    <a name="PodAffinityTerm"></a>
+    *Defines a set of pods (namely those matching the labelSelector relative to the given namespace(s)) that this pod should be co-located (affinity) or not co-located (anti-affinity) with, where co-located is defined as running on a node whose value of the label with key <topologyKey> matches that of any node on which a pod of the set of pods is running*
+  -->
+
+  - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm** (PodAffinityTerm)，必需
+
+    必需的字段。一个 Pod 亲和性条件，与相应的权重相关联。
```
Comment (tengqm) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3692]:
```suggestion
    必需的字段。一个 Pod 亲和性条件，对应一个与相应的权重值。
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 5]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+  The scheduler will prefer to schedule pods to nodes that satisfy the affinity expressions specified by this field, but it may choose a node that violates one or more of the expressions. The node that is most preferred is the one with the greatest sum of weights, i.e. for each node that meets all of the scheduling requirements (resource request, requiredDuringScheduling affinity expressions, etc.), compute a sum by iterating through the elements of this field and adding "weight" to the sum if the node has pods which matches the corresponding podAffinityTerm; the node(s) with the highest sum are the most preferred.
+
+  <a name="WeightedPodAffinityTerm"></a>
+  *The weights of all of the matched WeightedPodAffinityTerm fields are added per-node to find the most preferred node(s)*
+-->
+
+- **preferredDuringSchedulingIgnoredDuringExecution** ([]WeightedPodAffinityTerm)
+
+  调度器会更倾向于将 Pod 调度到满足该字段指定的亲和性表达式的节点，
+  但它可能会选择违反一个或多个表达式的节点。最优选的节点是权重总和最大的节点，
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3672]:
```suggestion
  但它可能会选择违反一个或多个表达式的节点。最优选择是权重总和最大的节点，
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 6]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+
+  <a name="PodAffinityTerm"></a>
+  *Defines a set of pods (namely those matching the labelSelector relative to the given namespace(s)) that this pod should be co-located (affinity) or not co-located (anti-affinity) with, where co-located is defined as running on a node whose value of the label with key <topologyKey> matches that of any node on which a pod of the set of pods is running*
+-->
+
+- **requiredDuringSchedulingIgnoredDuringExecution** （[]PodAffinityTerm）
+
+  如果在调度时不满足该字段指定的亲和性要求，则该 Pod 不会被调度到该节点上。
+  如果在 Pod 执行期间的某个时间点不再满足此字段指定的亲和性要求（例如：由于 Pod 标签更新），
+  系统可能会或可能不会尝试最终将 Pod 从其节点中逐出。
```
Comment (tengqm) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3771]:
```suggestion
  系统可能会也可能不会尝试最终将 Pod 从其节点中逐出。
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 7]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+  当有多个元素时，每个 `podAffinityTerm` 对应的节点列表是取其交集的，即必须满足所有条件。
```
Comment (tengqm) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3772]:
```suggestion
  当此列表中有多个元素时，每个 `podAffinityTerm` 对应的节点列表是取其交集的，即必须满足所有条件。
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 8]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+    *Defines a set of pods (namely those matching the labelSelector relative to the given namespace(s)) that this pod should be co-located (affinity) or not co-located (anti-affinity) with, where co-located is defined as running on a node whose value of the label with key <topologyKey> matches that of any node on which a pod of the set of pods is running*
+  -->
+
+  - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm** (PodAffinityTerm)，必需
+
+    必需的字段。一个 Pod 亲和性条件，与相应的权重相关联。
+
+    <a name="PodAffinityTerm"></a>
+    定义一组 Pod（即那些与给定名字空间相关的标签选择算符匹配的 Pod 集合），
+    当前 Pod 应该与所选 Pod 集合位于同一位置（亲和性）或不位于同一位置（反亲和性），
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3696]:
```suggestion
    当前 Pod 应该与所选 Pod 集合位于同一位置（亲和性）或位于不同位置（反亲和性），
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 9]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+    <!--
+    - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm.topologyKey** (string), required
+
+      This pod should be co-located (affinity) or not co-located (anti-affinity) with the pods matching the labelSelector in the specified namespaces, where co-located is defined as running on a node whose value of the label with key topologyKey matches that of any node on which any of the selected pods is running. Empty topologyKey is not allowed.
+    -->
+
+    - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm.topologyKey** (string)，必需
+
+      此 Pod 应与指定名字空间中与标签选择算符匹配的 Pod 集合位于同一位置（亲和性）
+      或不位于同一位置（反亲和性），这里的 "在同一位置" 意味着运行在一个节点上，其键名为
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3709]:
```suggestion
      或位于不同位置（反亲和性），这里的 "在同一位置" 意味着运行在一个节点上，其键名为
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 10]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+  *The weights of all of the matched WeightedPodAffinityTerm fields are added per-node to find the most preferred node(s)*
+-->
+
+- **preferredDuringSchedulingIgnoredDuringExecution** ([]WeightedPodAffinityTerm)
+
+  调度器会更倾向于将 Pod 调度到满足该字段指定的亲和性表达式的节点，
+  但它可能会选择违反一个或多个表达式的节点。最优选的节点是权重总和最大的节点，
+  即对于满足所有调度要求（资源请求、`requiredDuringScheduling` 亲和表达式等）的每个节点，
+  通过迭代该字段的元素来计算总和，如果节点具有与相应 `podAffinityTerm`
+  匹配的 Pod，则将 "权重" 添加到总和中；
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3675]:
```suggestion
  匹配的 Pod，则将“权重”添加到总和中； 
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 11]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+  -->
+
+  - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm** (PodAffinityTerm)，必需
+
+    必需的字段。一个 Pod 亲和性条件，与相应的权重相关联。
+
+    <a name="PodAffinityTerm"></a>
+    定义一组 Pod（即那些与给定名字空间相关的标签选择算符匹配的 Pod 集合），
+    当前 Pod 应该与所选 Pod 集合位于同一位置（亲和性）或不位于同一位置（反亲和性），
+    其中 "在同一位置" 意味着运行在一个节点上，其键 `topologyKey` 的标签值与运行所选 Pod
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3697]:
```suggestion
    其中“在同一位置”意味着运行在一个节点上，其键 `topologyKey` 的标签值与运行所选 Pod
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 12]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+    - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm.namespaceSelector** (<a href="{{< ref "../common-definitions/label-selector#LabelSelector" >}}">LabelSelector</a>)
+
+      A label query over the set of namespaces that the term applies to. The term is applied to the union of the namespaces selected by this field and the ones listed in the namespaces field. null selector and null or empty namespaces list means "this pod's namespace". An empty selector ({}) matches all namespaces.
+    -->
+
+    - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm.namespaceSelector** （<a href="{{< ref "../common-definitions/label-selector#LabelSelector" >}}">LabelSelector</a>）
+
+      对条件所适用的名字空间集合的标签查询。
+      此条件会被应用到此字段所选择的名字空间和 namespaces 字段中列出的名字空间的组合之上。
+      选择算符为 null 和 namespaces 列表为 null 值或空表示 "此 Pod 的名字空间"。
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3733]:
```suggestion
      选择算符为 null 和 namespaces 列表为 null 值或空表示“此 Pod 的名字空间”。
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 13]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+    - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm.namespaces** ([]string)
+
+      namespaces specifies a static list of namespace names that the term applies to. The term is applied to the union of the namespaces listed in this field and the ones selected by namespaceSelector. null or empty namespaces list and null namespaceSelector means "this pod's namespace".
+    -->
+
+    - **preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm.namespaces** （[]string）
+
+      namespaces 指定此条件所适用的名字空间，是一个静态列表。
+      此条件会被应用到 namespaces 字段中列出的名字空间和由 namespaceSelector 选中的名字空间上。
+      namespaces 列表为 null 或空，以及 namespaceSelector 值为 null 均表示 "此 Pod 的名字空间"。
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3746]:
```suggestion
      namespaces 列表为 null 或空，以及 namespaceSelector 值为 null 均表示“此 Pod 的名字空间”。
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 14]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+
+  如果在调度时不满足该字段指定的亲和性要求，则该 Pod 不会被调度到该节点上。
+  如果在 Pod 执行期间的某个时间点不再满足此字段指定的亲和性要求（例如：由于 Pod 标签更新），
+  系统可能会或可能不会尝试最终将 Pod 从其节点中逐出。
+  当有多个元素时，每个 `podAffinityTerm` 对应的节点列表是取其交集的，即必须满足所有条件。
+
+  <a name="PodAffinityTerm"></a>
+  定义一组 Pod（即那些与给定名字空间相关的标签选择算符匹配的 Pod 集合），当前 Pod 应该与该
+  Pod 集合位于同一位置（亲和性）或不位于同一位置（反亲和性）。
+  这里的 "位于同一位置" 含义是运行在一个节点上。基于 `topologyKey` 字段所给的标签键名，
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3777]:
```suggestion
  这里的“位于同一位置”含义是运行在一个节点上。基于 `topologyKey` 字段所给的标签键名，
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 15]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+  检查所选 Pod 集合中各个 Pod 所在的节点上的标签值，标签值相同则认作 "位于同一位置"。
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3778]:
```suggestion
  检查所选 Pod 集合中各个 Pod 所在的节点上的标签值，标签值相同则认作“位于同一位置”。
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 16]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+  - **requiredDuringSchedulingIgnoredDuringExecution.topologyKey** (string), required
+
+    This pod should be co-located (affinity) or not co-located (anti-affinity) with the pods matching the labelSelector in the specified namespaces, where co-located is defined as running on a node whose value of the label with key topologyKey matches that of any node on which any of the selected pods is running. Empty topologyKey is not allowed.
+  -->
+
+  - **requiredDuringSchedulingIgnoredDuringExecution.topologyKey** (string)，必需
+
+    此 Pod 应与指定名字空间中与标签选择算符匹配的 Pod 集合位于同一位置（亲和性）
+    或不位于同一位置（反亲和性），
+    这里的 "位于同一位置" 含义是运行在一个节点上。基于 `topologyKey` 字段所给的标签键名，
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3790]:
```suggestion
    这里的“位于同一位置”含义是运行在一个节点上。基于 `topologyKey` 字段所给的标签键名，
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 17]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+    检查所选 Pod 集合中各个 Pod 所在的节点上的标签值，标签值相同则认作 "位于同一位置"。
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3791]:
```suggestion
    检查所选 Pod 集合中各个 Pod 所在的节点上的标签值，标签值相同则认作“位于同一位置”。
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 18]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+  - **requiredDuringSchedulingIgnoredDuringExecution.namespaceSelector** (<a href="{{< ref "../common-definitions/label-selector#LabelSelector" >}}">LabelSelector</a>)
+
+    A label query over the set of namespaces that the term applies to. The term is applied to the union of the namespaces selected by this field and the ones listed in the namespaces field. null selector and null or empty namespaces list means "this pod's namespace". An empty selector ({}) matches all namespaces.
+  -->
+
+  - **requiredDuringSchedulingIgnoredDuringExecution.namespaceSelector** （<a href="{{< ref "../common-definitions/label-selector#LabelSelector" >}}">LabelSelector</a>）
+
+    对条件所适用的名字空间集合的标签查询。
+    当前条件将应用于此字段选择的名字空间和 namespaces 字段中列出的名字空间。
+    选择算符为 null 和 namespaces 列表为 null 或空值表示 “此 Pod 的名字空间”。
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3814]:
```suggestion
    选择算符为 null 和 namespaces 列表为 null 或空值表示“此 Pod 的名字空间”。
```

File: content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md [hunk 19]
```diff
@@ -3648,6 +3648,185 @@ Node affinity is a group of node affinity scheduling rules.
...
+  - **requiredDuringSchedulingIgnoredDuringExecution.namespaces** ([]string)
+
+    namespaces specifies a static list of namespace names that the term applies to. The term is applied to the union of the namespaces listed in this field and the ones selected by namespaceSelector. null or empty namespaces list and null namespaceSelector means "this pod's namespace".
+  -->
+
+  - **requiredDuringSchedulingIgnoredDuringExecution.namespaces** （[]string）
+
+    namespaces 指定当前条件所适用的名字空间名称的静态列表。
+    当前条件适用于此字段中列出的名字空间和由 namespaceSelector 选中的名字空间。
+    namespaces 列表为 null 或空，以及 namespaceSelector 为 null 表示 “此 Pod 的名字空间”。
```
Comment (howieyuen) [inline content/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1.md:3828]:
```suggestion
    namespaces 列表为 null 或空，以及 namespaceSelector 为 null 表示“此 Pod 的名字空间”。
```

**Review by howieyuen — CHANGES_REQUESTED** [review]

**Comment by Sea-n** [comment]
There are some suggestions marked as resolved without being fixed. Could you leave comments for them to explain how do you think?

**Comment by howieyuen** [comment]
> There are some suggestions marked as resolved without being fixed. Could you leave comments for them to explain how do you think?

I remember there was 20+ commits after @weiy730 modified first, so I leave the `tide/merge-method-squash` label.
I guest he wrongly used `git rebase` before `git push -f`.

**Comment by weiy730** [comment]
Sorry, and I fixed it.

**Comment by tengqm** [comment]
目前看，剩余几个没修改的问题都是关于使用全角引号的。请修订。

**Pull request merged**